docs/source/format/CanonicalExtensions.rst

```diff
@@ -283,6 +283,28 @@ UUID
    A specific UUID version is not required or guaranteed. This extension represents
    UUIDs as FixedSizeBinary(16) with big-endian notation and does not interpret the bytes in any way.
 
+8-bit Boolean
+=============
+
+Bool8 represents a boolean value using 1 byte (8 bits) to store each value instead of only 1 bit as in
+the native Arrow Boolean type. Although less compact that the native representation, Bool8 may have
+better zero-copy compatibility with various systems that also store booleans using 1 byte.
+
+* Extension name: ``arrow.bool8``.
+
+* The storage type of this extension is ``Int8`` where:
```

Comment:
Out of curiosity, is there a reason we prefer `int8` storage type to `uint8`?

Reply:
Hi @AlenkaF, either would work but `int8` was chosen for easier multi-language support. There are a few places in the Arrow docs that suggest this, such as the entry for [dictionary indices](https://arrow.apache.org/docs/format/Columnar.html#dictionary-encoded-layout):
> Since unsigned integers can be more difficult to work with in some cases (e.g. in the JVM), we recommend preferring signed integers over unsigned integers for representing dictionary indices. Additionally, we recommend avoiding using 64-bit unsigned integer indices unless they are required by an application.

The same principles will likely make `Bool8` work better on the JVM if it's represented with signed integers.

Reply:
Great to know, thank you!

Reply:
And since any negative value has at least 1 bit turned on, "non-zero means true" works for both signed and unsigned integers with 0 (false) has the same representation in both signed and unsigned integers.